### PR TITLE
Add Scale Value - rational constant multiplication.

### PR DESCRIPTION
**Pull request by nau**
Add Scale Value - rational constant multiplication with 'banking' rounding.

### CODE REVIEW

File: marlowe-playground-client/src/Marlowe/Semantics.purs [hunk 1]
```diff
@@ -223,12 +223,33 @@ instance hasArgsValueId :: Args ValueId where
   hasArgs _ = false
   hasNestedArgs _ = false
 
+data Rational
+  = Rational BigInteger BigInteger
+
+-- derive instance newtypeRational :: Newtype Rational _
```
Comment (shmish111) [inline marlowe-playground-client/src/Marlowe/Semantics.purs]: kill it!

File: marlowe-playground-client/src/Marlowe/Semantics.purs [hunk 2]
```diff
@@ -719,6 +740,36 @@ evalValue env state value =
...
+          if r == zero then
+            q
+          else
+            if sign == (-1) then
+              q
+            else
+              if sign == 0 && isEven then
+                q
+              else
+                m
```
Comment (shmish111) [inline marlowe-playground-client/src/Marlowe/Semantics.purs]: `if r == zero || sign == (-1) || (sign == 0 && isEven) then q else m` ?

File: marlowe-playground-client/grammar.ne [hunk 3]
```diff
@@ -83,6 +85,8 @@ number
     | %number {% ([n]) => opts.mkTerm(opts.mkBigInteger(n.value))({row: n.line, column: n.col}) %}
     | lparen %number rparen {% ([,n,]) => opts.mkTerm(opts.mkBigInteger(n.value))({row: n.line, column: n.col}) %}
 
+bigint -> %number {% ([n]) => opts.mkBigInteger(n.value) %}
+
```
[the remark this answers is not in the document]
Comment (nau) [reply marlowe-playground-client/grammar.ne]: It produces a Term rather than BigInteger

[the remark this answers is not in the document]
Comment (shmish111) [reply marlowe-playground-client/grammar.ne]: but is there a reason why you wouldn't want to use a term?

**Pull request merged**